internal/infrastructure/ceph/rbd_test.go

```diff
@@ -41,14 +43,16 @@ func TestApplyDiffToRawImage_success(t *testing.T) {
 	require.NoError(t, err)
 
 	gotFilePath := filepath.Join(tempDir, "test-raw.img")
-	err = applyDiffToRawImage(gotFilePath, openGZFile(t, "testdata/full.gz"), "", "snap20", 8*1024*1024)
+	fullReader := openGZFileWithChecksum(t, "testdata/full.gz", "testdata/full.csum", 2*1024*1024, false)
+	err = applyDiffToRawImage(gotFilePath, fullReader, "", "snap20", 8*1024*1024, 64*1024, 2*1024*1024, false)
 	assert.NoError(t, err)
 	fileInfo, err := os.Stat(gotFilePath)
 	assert.NoError(t, err)
 	assert.Equal(t, int64(16*1024*1024), fileInfo.Size())
 	utils.CompareReaders(t, openFileResized(t, gotFilePath, 100*1024*1024), openGZFile(t, "testdata/full-raw-img.gz"))
 
-	err = applyDiffToRawImage(gotFilePath, openGZFile(t, "testdata/diff.gz"), "snap20", "snap21", 8*1024*1024)
+	diffReader := openGZFileWithChecksum(t, "testdata/diff.gz", "testdata/diff.csum", 2*1024*1024, false)
+	err = applyDiffToRawImage(gotFilePath, diffReader, "snap20", "snap21", 8*1024*1024, 64*1024, 2*1024*1024, false)
 	assert.NoError(t, err)
 	utils.CompareReaders(t, openFileResized(t, gotFilePath, 100*1024*1024), openGZFile(t, "testdata/diff-raw-img.gz"))
```

Comment:
Let's check that the resulting checksum is correct. The current test writes the checksum file for the resulting raw image file, but it doesn't check that the checksum is correct.